lib/mutant/reporter/cli_compact/printer.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module Mutant
+  class Reporter
+    class CliCompact
```

Comment:
I suspect we have to make a base class for CLI reporters to reduce duplication between compact and "verbose". This way lots of  the private methods below get deduplicated.